### PR TITLE
Marathon Scheduler support for Authentication

### PR DESCRIPTION
**Pull request by jrcrawfo**
What?
Added support for sending along an authentication token with requests made to a Marathon instance

Why?
Some people have DC/OS clusters that sit behind an Oauth-based authentication system. In order to make topology submissions, a valid token must be provided

### CODE REVIEW

File: website/content/docs/operators/deployment/schedulers/dcos.md [hunk 1]
```diff
@@ -95,11 +100,15 @@ heron.directory.conf:                       "./heron-conf/"
...
+# The token of the marathon scheduler
+heron.marathon.scheduler.auth.token: "<auth_token>"
+
 # Invoke the IScheduler as a library directly
 heron.scheduler.is.service:                  False
 
 # location of the core package
-heron.package.core.uri:                      https://github.com/twitter/heron/releases/download/0.14.7/heron-core-0.14.7-ubuntu.tar.gz
+heron.package.core.uri:  
+https://github.com/twitter/heron/releases/download/0.14.7/heron-core-0.14.7-ubuntu.tar.gz
```
Comment (nlu90) [inline website/content/docs/operators/deployment/schedulers/dcos.md]: is this new line expected?

File: heron/schedulers/tests/java/com/twitter/heron/scheduler/marathon/MarathonControllerTest.java [hunk 2]
```diff
@@ -34,14 +34,16 @@
 @PrepareForTest(NetworkUtils.class)
 public class MarathonControllerTest {
   private static final String MARATHON_URI = "http://marathon.uri:8080";
+  private static final String MARATHON_AUTH_TOKEN = null;
```
Comment (nlu90) [inline heron/schedulers/tests/java/com/twitter/heron/scheduler/marathon/MarathonControllerTest.java]: could you add a test case for the auth failure?

**Comment by kramasamy** [comment]
@jrcrawfo - looks good to me. Can you please address @nlu90's feedback so that we can merge it?

File: heron/schedulers/tests/java/com/twitter/heron/scheduler/marathon/MarathonControllerTest.java (hunk 2 above)
Comment (jrcrawfo) [inline heron/schedulers/tests/java/com/twitter/heron/scheduler/marathon/MarathonControllerTest.java]: Yes, I can try to add a test here where i'm looking for the specific response code

File: website/content/docs/operators/deployment/schedulers/dcos.md (hunk 1 above)
Comment (jrcrawfo) [inline website/content/docs/operators/deployment/schedulers/dcos.md]: Nope, will reset that

**Comment by objmagic** [comment]
@jrcrawfo @kramasamy Is this the one you want to include in the next release? If so, please address comments asap.

**Comment by kramasamy** [comment]
@jrcrawfo - please address the comments of @nlu90 so that we can merge it.

**Comment by jrcrawfo** [comment]
@kramasamy @nlu90 Added the test and fixed the new line

**Comment by objmagic** [comment]
@jrcrawfo merge with master to fix CI error
@nlu90 can you review it? @kramasamy wants in next release.

**Pull request merged**